src/Select/index.js

```diff
@@ -123,21 +130,26 @@ const Select = ({
 }) => {
   let items = [];
   let categories = [];
+  let actions = [];
   const [userInput, setUserInput] = useState(""); // most recent val the user typed while focused on this input
 
-  const allChildren = React.Children.toArray(children);
+  const options = React.Children.toArray(children).filter(
+    (item) => !isAction(item),
+  );
+  actions = React.Children.toArray(children).filter(isAction);
```

Comment:
Now we _always_ extract `actions` as separate from `options`. You can see below that when we assign the `items` var, it's a combination of `[...options,...actions]`. This array we set to the `items` var is essential to downshift state management.

Regardless of how and where we render these things, downshift needs to have a complete accounting of each item. Rendering matches the item order up with downshift in `renderItem`, which uses prop getters to apply the appropriate props to each item.